src/network/networkctl.c

```diff
@@ -268,12 +275,29 @@ static int list_links(int argc, char *argv[], void *userdata) {
 
                 t = link_get_type_string(links[i].iftype, d);
 
-                printf("%3i %-16s %-18s %s%-16s%s %s%-10s%s\n",
-                       links[i].ifindex, links[i].name, strna(t),
-                       on_color_operational, strna(operational_state), off_color_operational,
-                       on_color_setup, strna(setup_state), off_color_setup);
+                r = table_add_cell_full(table, NULL, TABLE_UINT32, (uint32_t *) &links[i].ifindex, (size_t) -1, (size_t) -1, 0, 100, 0);
+                if (r < 0)
+                        return r;
+
+                r = table_add_many(table,
+                                   TABLE_STRING, links[i].name,
+                                   TABLE_STRING, strna(t),
+                                   TABLE_STRING, strna(operational_state),
+                                   TABLE_STRING, strna(setup_state));
+                if (r < 0)
+                        return log_error_errno(r, "Failed to add table cells: %m");
+
+                assert_se(cell = table_get_cell(table, i+1, 3));
+                assert_se(table_set_color(table, cell, on_color_operational) >= 0);
```

Comment:
table_set_color() can fail due to OOM. It's fine to ignore the result (by casting explicitly to `(void)`), given that coloring is not that important. However, we really shouldn't use `assert_se()` here I am sure, as that would mean we abort on OOM and I don't think we should...